hail/python/hailtop/batch/batch.py

```diff
@@ -149,6 +151,64 @@ def __init__(self,
 
         self._cancel_after_n_failures = cancel_after_n_failures
 
+        self._python_function_defs: Dict[int, Callable] = {}
+        self._python_function_files: Dict[int, Optional[_resource.InputResourceFile]] = {}
```

Comment:
Is `None` ever a valid value in this dictionary? From how it's used below I think if a key exists in the dictionary it'll point to a valid `InputResourceFile`